Dependency issue : go: github.com/jetstack/cert-manager@v0.12.0 requires launchpad.net/gocheck

**Describe the problem**:

When we import the new version - v0.12.0 of the go module of the `certmanager` within the `go.mod` file, then the following `require` message is displayed when dependencies are checked

```
go: github.com/jetstack/cert-manager@v0.12.0 requires
        launchpad.net/gocheck@v0.0.0-20140225173054-000000000087: bzr branch --use-existing-dir -- https://launchpad.net/~niemeyer/gocheck/trunk . in /Users/dabou/Code/go-workspace/pkg/mod/cache/vcs/f46ce2ae80d31f9b0a29099baa203e3b6d269dace4e5357a2cf74bd109e13339: exec: "bzr": executable file not found in $PATH
make: *** [verify] Error 1
```

Questions: 
- Can we make this module `launchpad.net/gocheck` optional to avoid to have to import it ?
- Why do we need the `bzr` executable ?

**Additional info**:
- Webhook godaddy project : https://github.com/snowdrop/godaddy-webhook
- Go mod file used
```
module github.com/snowdrop/godaddy-webhook

go 1.12

require (
	github.com/ghodss/yaml v0.0.0-20180820084758-c7ce16629ff4 // indirect
	github.com/jetstack/cert-manager v0.12.0
	github.com/natefinch/lumberjack v2.0.0+incompatible // indirect
	gopkg.in/yaml.v1 v1.0.0-20140924161607-9f9df34309c0 // indirect
	k8s.io/apiextensions-apiserver v0.0.0-20191114105449-027877536833
	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
)

replace k8s.io/client-go => k8s.io/client-go v0.0.0-20190413052642-108c485f896e
```

**Environment details:**:
- cert-manager version (e.g. v0.4.0): 0.12.0

/kind bug

If I install the `bzr` or `bazaar` tool using `brew install bzr`, then I egt another issue now
```
 make
go test -v .
# k8s.io/component-base/metrics/legacyregistry
../../go-workspace/pkg/mod/k8s.io/component-base@v0.0.0-20191114102325-35a9586014f7/metrics/legacyregistry/registry.go:44:9: undefined: prometheus.InstrumentHandler
# k8s.io/apiserver/pkg/util/webhook
../../go-workspace/pkg/mod/k8s.io/apiserver@v0.0.0-20191114103151-9ca1dc586682/pkg/util/webhook/client.go:143:6: cfg.NextProtos undefined (type *rest.Config has no field or method NextProtos)
FAIL    github.com/snowdrop/godaddy-webhook [build failed]
FAIL
make: *** [verify] Error 2

```

using this go.mod definition
```
module github.com/snowdrop/godaddy-webhook

go 1.12

require (
	github.com/jetstack/cert-manager v0.12.0
	k8s.io/apiextensions-apiserver v0.0.0-20191114105449-027877536833
	k8s.io/client-go v11.0.1-0.20190409021438-1a26190bd76a+incompatible
)

replace k8s.io/client-go => k8s.io/client-go v0.0.0-20190413052642-108c485f896e
```

If I use the following modules
```
module github.com/snowdrop/godaddy-webhook

go 1.13

require (
require (
	github.com/jetstack/cert-manager v0.12.0
	k8s.io/apiextensions-apiserver v0.0.0-20191114105449-027877536833
	k8s.io/client-go v0.0.0-20191114101535-6c5935290e33
	k8s.io/component-base v0.0.0-20191114102325-35a9586014f7
)
```
then I get this error `../../go-workspace/pkg/mod/k8s.io/component-base@v0.0.0-20191114102325-35a9586014f7/metrics/legacyregistry/registry.go:44:9: undefined: prometheus.InstrumentHandler`
```
make compile
CGO_ENABLED=0 go build -o webhook -ldflags '-w -extldflags "-static"' .
# k8s.io/component-base/metrics/legacyregistry
../../go-workspace/pkg/mod/k8s.io/component-base@v0.0.0-20191114102325-35a9586014f7/metrics/legacyregistry/registry.go:44:9: undefined: prometheus.InstrumentHandler

```

--- a/LICENSES
+++ b/LICENSES
@@ -11890,41 +11890,6 @@ SOFTWARE.
 
 
 ================================================================================
-= vendor/github.com/tent/http-link-go licensed under: =
-
-Copyright (c) 2013 Tent.is, LLC. All rights reserved.
-
-Redistribution and use in source and binary forms, with or without
-modification, are permitted provided that the following conditions are
-met:
-
-   * Redistributions of source code must retain the above copyright
-notice, this list of conditions and the following disclaimer.
-   * Redistributions in binary form must reproduce the above
-copyright notice, this list of conditions and the following disclaimer
-in the documentation and/or other materials provided with the
-distribution.
-   * Neither the name of Tent.is, LLC nor the names of its
-contributors may be used to endorse or promote products derived from
-this software without specific prior written permission.
-
-THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
-"AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
-LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
-A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
-OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
-SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
-LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
-DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
-THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
-(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
-OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
-
-= vendor/github.com/tent/http-link-go/LICENSE 8a6182596e752af2ca1f626050072444
-================================================================================
-
-
-================================================================================
 = vendor/github.com/Venafi/vcert licensed under: =
 
                                  Apache License

--- a/go.mod
+++ b/go.mod
@@ -17,11 +17,10 @@ require (
 	github.com/aws/aws-sdk-go v1.24.1
 	github.com/cloudflare/cloudflare-go v0.8.5
 	github.com/cpu/goacmedns v0.0.0-20180701200144-565ecf2a84df
-	github.com/digitalocean/godo v1.6.0
+	github.com/digitalocean/godo v1.29.0
 	github.com/go-logr/logr v0.1.0
 	github.com/go-logr/zapr v0.1.1 // indirect
 	github.com/google/go-github v17.0.0+incompatible
-	github.com/google/go-querystring v1.0.0 // indirect
 	github.com/google/gofuzz v1.0.0
 	github.com/gorilla/context v1.1.1 // indirect
 	github.com/gorilla/mux v1.6.2
@@ -38,7 +37,6 @@ require (
 	github.com/spf13/cobra v0.0.5
 	github.com/spf13/pflag v1.0.5
 	github.com/stretchr/testify v1.4.0
-	github.com/tent/http-link-go v0.0.0-20130702225549-ac974c61c2f9 // indirect
 	golang.org/x/crypto v0.0.0-20191202143827-86a70503ff7e
 	golang.org/x/net v0.0.0-20191004110552-13f9640d40b9
 	golang.org/x/oauth2 v0.0.0-20190604053449-0f29369cfe45
@@ -55,7 +53,6 @@ require (
 	k8s.io/kube-aggregator v0.17.0
 	k8s.io/kube-openapi v0.0.0-20191107075043-30be4d16710a
 	k8s.io/utils v0.0.0-20191114184206-e782cd3c129f
-	launchpad.net/gocheck v0.0.0-20140225173054-000000000087 // indirect
 	sigs.k8s.io/controller-runtime v0.3.1-0.20191022174215-ad57a976ffa1
 	sigs.k8s.io/controller-tools v0.2.2
 	sigs.k8s.io/testing_frameworks v0.1.1

--- a/hack/build/repos.bzl
+++ b/hack/build/repos.bzl
@@ -342,8 +342,8 @@ def go_repositories():
         build_file_generation = "on",
         build_file_proto_mode = "disable",
         importpath = "github.com/digitalocean/godo",
-        sum = "h1:BjFbURUz1LDimGfApIWqzZiSrJFbtyGYOHfSRq/Voi8=",
-        version = "v1.6.0",
+        sum = "h1:KgNNU0k9SZqVgn7m8NN9iDsq0+nluHBe8HR9QE0QVmA=",
+        version = "v1.29.0",
     )
     go_repository(
         name = "com_github_docker_docker",

